File: TP3/Models/Livre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.livres.model;

/**
 *
 * @author usager
 */
public class Livre {
    
    private String ISBN;
    private String Author;
    private String Title;
    private int NB_Pages;
    private String Edition;
    private int Year;
    private String Language;
    private String Description;
    private String Keywords;

    public void setKeywords(String Keywords) {
        this.Keywords = Keywords;
    }

    public String getKeywords() {
        return Keywords;
    }
    
    public void setISBN(String ISBN) {
        this.ISBN = ISBN;
    }

    public void setAuthor(String Author) {
        this.Author = Author;
    }

    public void setTitle(String Title) {
        this.Title = Title;
    }

    public void setNB_Pages(int NB_Pages) {
        this.NB_Pages = NB_Pages;
    }

    public void setEdition(String Edition) {
        this.Edition = Edition;
    }

    public void setYear(int Year) {
        this.Year = Year;
    }

    public void setLanguage(String Languge) {
        this.Language = Languge;
    }

    public void setDescription(String Description) {
        this.Description = Description;
    }

    public String getISBN() {
        return ISBN;
    }

    public String getAuthor() {
        return Author;
    }

    public String getTitle() {
        return Title;
    }

    public int getNB_Pages() {
        return NB_Pages;
    }

    public String getEdition() {
        return Edition;
    }

    public int getYear() {
        return Year;
    }

    public String getLanguage() {
        return Language;
    }

    public String getDescription() {
        return Description;
    }
    
}

File: TP3/Models/Livre.cs
using System.ComponentModel.DataAnnotations;

namespace TP3.Models
{
    public class Livre
    {
        [Required(ErrorMessage = "Un isbn et requis")]
        public string isbn { get; set; }
        [Required(ErrorMessage = "Un auteur et requis")]
        public string author { get; set; }
        [Required(ErrorMessage = "Un title et requis")]
        public string title { get; set; }
        [Required(ErrorMessage = "Un nombre de pages et requis")]
        public int nbPages { get; set; }
        [Required(ErrorMessage = "Une edition et requise")]
        public string edition { get; set; }
        [Required(ErrorMessage = "Une annee et requise"), Range(-9999,9999)]
        public int year { get; set; }
        [Required(ErrorMessage = "Une langue et requise")]
        public string language { get; set; }
        [Required(ErrorMessage = "Une description et requise")]
        public string description { get; set; }
        [Required(ErrorMessage = "Un ou des keywords sont requis")]
        public string keywords { get; set; }

        public Livre(string isbn, string author, string title, int nbPages, string edition, int year, string language, string description, string keywords)
        {
            this.isbn = isbn;
            this.author = author;
            this.title = title;
            this.nbPages = nbPages;
            this.edition = edition;
            this.year = year;
            this.language = language;
            this.description = description;
            this.keywords = keywords;
        }
    }
}
